### PR TITLE
Remove query and site props passing from everywhere except class components

### PR DESCRIPTION
**Pull request by apata**
### Changes

Followup to https://github.com/plausible/analytics/pull/4334, to further alleviate issues from props passing

### Tests
- [ ] Automated tests have been added
- [ ] This PR does not require tests

### Changelog
- [ ] Entry has been added to changelog
- [x] This PR does not make a user-facing change

### Documentation
- [ ] [Docs](https://github.com/plausible/docs) have been updated
- [x] This change does not need a documentation update

### Dark mode
- [ ] The UI has been tested both in dark and light mode
- [x] This PR does not change the UI

### CODE REVIEW

File: assets/js/dashboard/comparison-input.js [hunk 1]
```diff
@@ -86,7 +86,9 @@ function ComparisonModeOption({ label, value, isCurrentlySelected, updateMode, s
   )
 }
 
-function MatchDayOfWeekInput({ history, query, site }) {
+function MatchDayOfWeekInput({ history }) {
+  const site = useSiteContext()
+  const { query } = useQueryContext()
```
Comment (macobo) [inline assets/js/dashboard/comparison-input.js:91]: Would something like `useDashboardContext()` make sense which would encapsulate both site AND query? :thinking:

File: assets/js/dashboard/stats/imported-query-unsupported-warning.js [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import React from "react";
 import { ExclamationCircleIcon } from '@heroicons/react/24/outline'
 import FadeIn from "../fade-in";
+import { useQueryContext } from "../query-context";
```
[the remark this answers is not in the document]
Comment (apata) [reply assets/js/dashboard/stats/imported-query-unsupported-warning.js:4]: Thanks! Agreed, but I can't get prettier and eslint to work in the repository / in my IDE well. Not sure yet what the issue is with those, will come back to that if the eslint updates that I need to do for Typescript anyways don't fix it.

File: assets/js/dashboard/comparison-input.js (hunk 1 above)
Comment (apata) [reply assets/js/dashboard/comparison-input.js:91]: Thanks!

I considered it initially, but my hunch is that many of these components don't really need either. 

I think the way forward is to actually remove some of these superfluous dependencies, and provide components with specific handlers for what they're supposed to do: in this example handleCompareExactMatch, handleCompareDayOfTheWeek. 

Such handlers would be declared higher up in the React tree and themselves dependent on a shared API component, which is dependent on the site.domain.

**Review by macobo — APPROVED** [review]
Didn't test it manually but the approach and code looks great! :ship:

**Pull request merged**